homeassistant/components/homekit_controller/climate.py

```diff
@@ -170,8 +178,49 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
             CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.ROTATION_SPEED,
         ]
 
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
+            return [FAN_OFF,FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        speed = self.service.value(CharacteristicsTypes.ROTATION_SPEED)
+        fan_mode = FAN_OFF
+       
+        #homekit seems to set value 0 33 66 100
```

Comment:
We have utilities for converting fan speeds already. There are examples in the fan base entity and fan development docs

Reply:
used the utils